pkg/networkutils/network.go

```diff
@@ -338,8 +338,9 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDRs []string, primaryMAC string, pr
 			}})
 	}
 
-	// Prepare the Desired Rule for SNAT Rule
-	snatRule := []string{"-m", "comment", "--comment", "AWS, SNAT",
+	// Prepare the Desired Rule for SNAT Rule for non-pod ENIs
+	snatRule := []string{"!", "-o", "vlan+",
+		"-m", "comment", "--comment", "AWS, SNAT",
```

Comment:
This is important, we can't SNAT traffic from pods with Branch ENIs.

Reply:
Right, what does this affect?  Does this mean we can't send traffic outside the immediate VPC from these pods? (that seems bad)


Aside: my quick search just turned up https://docs.aws.amazon.com/eks/latest/userguide/external-snat.html which says:
> SNAT is necessary because the internet gateway only knows how to translate between the primary private and public or elastic IP address assigned to the primary elastic network interface of the Amazon EC2 instance node that pods are running on.

Huh, I wouldn't have guessed this limitation, and I couldn't find more info in the VPC docs after a brief search.  Is this an IGW limitation, or some side effect of routes that we configure for secondary addresses within CNI?  (as in: how will this affect vlan interfaces, if they are configured with the vpc subnet gateway?)

Reply:
Ingress and Egress permission on the security group is applied to the traffic outside the instance, which means that we can't do any NAT within the instance for these pods. This will bypass the security group check.

If these pods uses private subnet which has NAT gateway associated then traffic will follow out to the internet. Otherwise it can talk only to resources within the VPC.

Reply:
Thanks, that makes sense.  Does that mean we could do the same (NAT at ngw) for secondary IPs, and remove the iptables snat?

Reply:
Yes we can remove if worker nodes subnets have NAT attached, then we don't have to do this SNAT at the host.

Reply:
If they have a NAT Gateway, they can set the wonderfully named `AWS_VPC_K8S_CNI_EXTERNALSNAT=true`, and then `aws-node` won't set up any SNAT rules ([Through this tricky code...](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/networkutils/network.go#L314)).